File: Unidad I/PAC III 2019/Tarea N1/Python/Stack.py
class Node:
    def __init__(self,value):
        self.value = value
        self.prev = None

class Stack:
    def __init__(self):
        self.last = None

    def push(self,value):
        if not self.last:
            self.last = Node(value)
            return True
        else:
            before = self.last
            self.last = Node(value)
            self.last.prev = before
            return True

    def printS(self):
        if not self.last:
            return None
        else:
            current = self.last
            stack = ("%s%s" %("\n",current.value))
            while current.prev:
                current = current.prev
                stack = stack + ("%s%s" %("\n",current.value))
            return stack

    def pop(self):
        if not self.last:
            return None
        else:
            if not self.last.prev:
                temp = self.last
                self.last = None
                return temp.value
            else:
                current = self.last
                self.last = current.prev
                return current.value
        return None

File: Unidad I/PAC III 2019/Tarea N1/JS/Stack.js
function Stack() {
    this.last = null;
    this.push = StackPush;
    this.print = StackPrint;
    this.pop = StackPop;
}

function StackPush(value) {
    if (!this.last) {
        this.last = new Node(value);
        return true;
    } else {
        prev = this.last;
        this.last = new Node(value);
        this.last.prev = prev;
        return true;
    }
}

function StackPrint() {
    stack = "\n";
    current = this.last;
    while (current) {
        stack += current.value.taste;
        stack += "\n";
        current = current.prev;
    }
    return stack;
}

function StackPop() {
    if (!this.last) {
        return false;
    } else {
        if (!this.last.prev) {
            temp = this.last;
            this.last = null;
            return temp.value.taste;
        } else {
            current = this.last;
            this.last = current.prev;
            return current.value.taste;
        }
    }
    return false;
}